### PR TITLE
Fix various Linux issues related to the installer and to the runner.

### PR DESCRIPTION
**Pull request by niezbop**
Fixes various small bugs left in the installer and in the runner under Linux.

### CODE REVIEW

File: lib/u3d/installer.rb [hunk 1]
```diff
@@ -170,7 +172,7 @@ def self.create
                   end
       if UI.interactive?
         unclean = []
-        installer.installed.each { |unity| unclean << unity unless unity.path =~ UNITY_DIR_CHECK }
+        installer.installed.each { |unity| unclean << unity unless clean_install?(unity.path) }
```
Comment (lacostej) [inline lib/u3d/installer.rb]: I would prefer this to be implemented as `unity.clean_install?` I think and keep each class knowledgeable about it.

File: lib/u3d/installer.rb [hunk 2]
```diff
@@ -384,7 +397,70 @@ def exist?
...
+    def self.install_dependencies
+      if `which dpkg` != ''
+        prefix = 'apt-get -y install'
+      elsif `which rpm` != ''
+        prefix = 'yum -y install'
+      else
+        raise 'Cannot install dependencies on your Linux distribution'
+      end
+      DEPENDENCIES.each do |dep|
+        if UI.interactive?
```
Comment (lacostej) [inline lib/u3d/installer.rb]: we want to ask only once

File: lib/u3d/installer.rb [hunk 3]
```diff
@@ -384,7 +397,70 @@ def exist?
...
+      if Helper.linux?
+        version.gsub!(/Linux/, '')
+        version.gsub!(/x/, '')
+      end
+      version
+    end
+  end
+
+  class LinuxDependencies
+    DEPENDENCIES = [
```
Comment (lacostej) [inline lib/u3d/installer.rb]: are those dependencies the same for all versions?

File: lib/u3d/installer.rb [hunk 4]
```diff
@@ -384,7 +397,70 @@ def exist?
...
+      elsif `which rpm` != ''
+        prefix = 'yum -y install'
+      else
+        raise 'Cannot install dependencies on your Linux distribution'
+      end
+      DEPENDENCIES.each do |dep|
+        if UI.interactive?
+          next unless UI.confirm "Install #{dep}?"
+        end
+        U3dCore::CommandExecutor.execute(command: "#{prefix} #{dep}", admin: true)
```
Comment (lacostej) [inline lib/u3d/installer.rb]: we can probably install all deps in a single command.
command: "#{prefix} #{DEPENDENCIES.join(' ')}"

File: lib/u3d/installer.rb [hunk 5]
```diff
@@ -384,7 +397,70 @@ def exist?
     def editor_version
       require 'yaml'
       yaml = YAML.load(File.read("#{@path}/ProjectSettings/ProjectVersion.txt"))
-      yaml['m_EditorVersion']
+      version = yaml['m_EditorVersion']
+      if Helper.linux?
+        version.gsub!(/Linux/, '')
```
Comment (lacostej) [inline lib/u3d/installer.rb]: I would prefer a more explicit matcher than using `gsub!()`

File: lib/u3d/installer.rb (hunk 4 above)
Comment (niezbop) [inline lib/u3d/installer.rb]: We could be I still wanted to let the user specify which packages he wanted - or didn't want - to install for whatever reason. Hence the each loop and the UI.confirm

File: lib/u3d/installer.rb (hunk 3 above)
Comment (niezbop) [inline lib/u3d/installer.rb]: I'm not sure the [blogpost](https://forum.unity3d.com/threads/unity-on-linux-release-notes-and-known-issues.350256/) presents the  dependencies alongside the first release, but the dependencies are no longer mentionned in this thread, which is supposed to be the official one.

File: lib/u3d/installer.rb (hunk 4 above)
Comment (niezbop) [inline lib/u3d/installer.rb]: Just saw previous comment about only asking one. Ignore previous comment

**Comment by niezbop** [comment]
Main PR closed => split into #25 #24 #26

**Pull request closed without merging**